website/blog/2025-12-11-autoscale-inference-workloads-with-kaito/index.md

```diff
@@ -0,0 +1,158 @@
+---
+title: "Autoscale KAITO inference workloads on AKS using KEDA"
+date: "2025-12-11"
+description: "Autoscale your KAITO inference workloads using KEDA"
+authors: ["andy-zhang"]
+tags: ["ai", "kaito"]
+---
+
+[Kubernetes AI Toolchain Operator](https://github.com/Azure/kaito/tree/main) (KAITO) is an operator that automates the AI/ML model inference or tuning workload in a Kubernetes cluster. With the [v0.8.0 release](https://github.com/Azure/kaito/releases/tag/v0.8.0), KAITO has introduced intelligent autoscaling for inference workloads as an alpha feature.
+
+## Overview
+
+This blog outlines the steps to enable intelligent autoscaling based on the service monitoring metrics for KAITO inference workloads by using the following components and features:
+
+- [KEDA](https://github.com/kedacore/keda)
+
+  - Kubernetes-based Event Driven Autoscaling component
+- [keda-kaito-scaler](https://github.com/kaito-project/keda-kaito-scaler)
+  - A dedicated KEDA external scaler, eliminating the need for external dependencies such as Prometheus.
+- KAITO `InferenceSet` CRD and Controller
+  - This new CRD and Controller were built on top of the KAITO workspace for intelligent autoscaling, introduced as an alpha feature in KAITO version `v0.8.0`
+
+### Architecture
+
+ ![keda-kaito-scaler-arch](keda-kaito-scaler-arch.png)
+
+## Prerequisites
+
+- install KEDA
+
+> The following example demonstrates how to install KEDA using Helm chart. For instructions on installing KEDA through other methods, please refer to the guide [deploying-keda](https://github.com/kedacore/keda#deploying-keda).
+
+```bash
+helm repo add kedacore https://kedacore.github.io/charts
+helm install keda kedacore/keda --namespace keda --create-namespace
+```
+
+- install keda-kaito-scaler
+
+```bash
+helm repo add keda-kaito-scaler https://kaito-project.github.io/keda-kaito-scaler/charts/kaito-project
+helm upgrade --install keda-kaito-scaler -n kaito-workspace keda-kaito-scaler/keda-kaito-scaler --create-namespace
+```
+
```

Comment:
Just have a check: whether both keda and keda-kaito-scaler should be deployed in the same namespace?

Reply:
only keda-kaito-scaler should be deployed in kaito-workspace ns, keda could be deployed on any namespace